.github/workflows/build-debug-node.yml

```diff
@@ -0,0 +1,54 @@
+name: Build debug node 
+
+on: 
+  workflow_dispatch:
+    inputs:
+      version:
+        required: true
+        description: 'Node.js version'
+
+jobs:
+  build:
+    name: Build Debug version of Node.js
+    runs-on: buildjet-4vcpu-ubuntu-2204
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Install dependencies 
+        run: apt-get install python3 g++ make python3-pip
+
+      - name: Download Node.js source
+        uses: actions/checkout@v4
+        with: 
+          repository: 'nodejs/node'
+          ref: 'v${{ github.event.inputs.version }}'
+          path: 'nodejs'
+
+      - name: Configure nodejs with debug flag 
+        run: ./configure --debug
+        working-directory: 'nodejs'
+
+      - name: Compile the nodejs
+        run: make -j4
+        working-directory: 'nodejs'
+
+      - name: Verify the build 
+        run: make test-only 
+        working-directory: 'nodejs'
+
+      - name: Create destination folder 
+        run: mkdir -p ${{ github.workspace }}/nodejs-debug-build-${{ github.event.inputs.version }}
+
+      - name: Copy nodejs build
+        run: make install 
+        working-directory: 'nodejs'
+        env: 
+          DESTDIR: ${{ github.workspace }}/nodejs-debug-build-${{ github.event.inputs.version }}
```

Comment:
Running `make install` only moves the `Release` version to the `DESTDIR`.  

https://github.com/nodejs/node/blob/609cd7f5bf6c1b2373353fb3fcaed2a3bbe1efa7/tools/install.py#L146

You need to also copy `nodejs/out/Debug/node` binary over the `Release` binary in `DESTDIR/lib/node` (pretty sure about that dest path but not 1000%)